pyfr/plugins/ascent.py

```diff
@@ -0,0 +1,290 @@
+# -*- coding: utf-8 -*-
+
+from collections import defaultdict
+import numpy as np
+import re
+
+from pyfr.inifile import _ensure_float
+from pyfr.libs.ascent import Ascent
+from pyfr.libs.conduit import ConduitNode
+from pyfr.mpiutil import get_comm_rank_root
+from pyfr.nputil import npeval
+from pyfr.plugins.base import BasePlugin
+from pyfr.shapes import BaseShape
+from pyfr.util import file_index, file_path, subclass_where
+from pyfr.writers.vtk import BaseShapeSubDiv
+
+
+class AscentPlugin(BasePlugin):
+    name = 'ascent'
+    systems = ['*']
+    formulations = ['dual', 'std']
+
+    def __init__(self, intg, cfgsect, suffix=None):
+        super().__init__(intg, cfgsect, suffix)
+
+        comm, rank, root = get_comm_rank_root()
+        self.nsteps = self.cfg.getint(cfgsect, 'nsteps')
+
+        # Underlying system
+        system = intg.system
+        self._ele_map = system.ele_map
+        self.dtype = intg.system.backend.fpdtype
+
+        # Underlying system elements class
+        self.elementscls = system.elementscls
+
+        # Expressions to plot and configs
+        cons = self.cfg.items_as('constants', float)
+        self._expr = {}
+        self._scene_cfg = self._config_scene(cfgsect, cons)
+        self._pipel_cfg = self._config_pipeline(cfgsect)
+        
+        # Gradient pre-processing
+        self._init_gradients(intg)
+
+        # Base directory for output and counter reset
+        self.basedir = self.cfg.getpath(cfgsect, 'basedir', '.', abs=True)
+        self.nout = {}
+        for sname in self._scene_cfg:
+            name = self._scene_cfg[sname]['image_name']
+            self.nout[name] = file_index(self.basedir, name) if intg.isrestart else 0
+
+        # Set subdivision
+        self.divisor = self.cfg.getint(cfgsect, 'devision', 
+                                       self.cfg.getint('solver', 'order'))
+        self.etypes_div = defaultdict(lambda: self.divisor)
```

Comment:
This should not be needed (VTK needs it to handle issues with high-order pyramids).

Reply:
See previous remark.